### PR TITLE
chore: bump postgrest to v12.2.0

### LINKED ISSUE #2361: PostgREST 12.1 frequently stuck in "Attempting to connect to the database..." (Local Development)
**Describe the bug**

Supabase PostgREST docker container logs:

```bash
2024-06-03 16:12:18 03/Jun/2024:09:12:18 +0000: Starting PostgREST 12.1...
2024-06-03 16:12:18 03/Jun/2024:09:12:18 +0000: Attempting to connect to the database...
2024-06-03 16:12:18 03/Jun/2024:09:12:18 +0000: Admin server listening on port 3001
2024-06-03 16:12:18 03/Jun/2024:09:12:18 +0000: Listening on port 3000
2024-06-03 16:12:18 03/Jun/2024:09:12:18 +0000: Listening for notifications on the "pgrst" channel
2024-06-03 16:12:18 03/Jun/2024:09:12:18 +0000: Successfully connected to PostgreSQL 15.1 (Ubuntu 15.1-1.pgdg20.04+1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0, 64-bit
2024-06-03 16:12:18 03/Jun/2024:09:12:18 +0000: Config reloaded
2024-06-03 16:12:19 03/Jun/2024:09:12:18 +0000: Schema cache queried in 7.2 milliseconds
2024-06-03 16:12:19 03/Jun/2024:09:12:18 +0000: Schema cache loaded 87 Relations, 83 Relationships, 48 Functions, 0 Domain Representations, 4 Media Type Handlers
2024-06-03 16:17:57 03/Jun/2024:09:17:57 +0000: Starting PostgREST 12.1...
2024-06-03 16:17:57 03/Jun/2024:09:17:57 +0000: Attempting to connect to the database...
2024-06-03 16:17:57 03/Jun/2024:09:17:57 +0000: Admin server listening on port 3001
2024-06-03 16:17:57 03/Jun/2024:09:17:57 +0000: Listening on port 3000
2024-06-03 16:17:57 Assertion failed: *(&once->n) != 4 (threads.c: k5_once: 564)
2024-06-03 16:17:57 03/Jun/2024:09:17:57 +0000: Starting PostgREST 12.1...
2024-06-03 16:17:57 03/Jun/2024:09:17:57 +0000: Attempting to connect to the database...
2024-06-03 16:17:57 03/Jun/2024:09:17:57 +0000: Admin server listening on port 3001
2024-06-03 16:17:57 03/Jun/2024:09:17:57 +0000: Listening on port 3000
2024-06-03 16:17:57 Assertion failed: *(&once->n) != 4 (threads.c: k5_once: 564)
2024-06-03 16:17:58 03/Jun/2024:09:17:58 +0000: Starting PostgREST 12.1...
2024-06-03 16:17:58 03/Jun/2024:09:17:58 +0000: Attempting to connect to the database...
2024-06-03 16:17:58 03/Jun/2024:09:17:58 +0000: Admin server listening on port 3001
2024-06-03 16:17:58 03/Jun/2024:09:17:58 +0000: Listening on port 3000
2024-06-03 16:17:58 Assertion failed: *(&once->n) != 4 (threads.c: k5_once: 564)
2024-06-03 16:17:59 03/Jun/2024:09:17:59 +0000: Starting PostgREST 12.1...
2024-06-03 16:17:59 03/Jun/2024:09:17:59 +0000: Attempting to connect to the database...
2024-06-03 16:17:59 03/Jun/2024:09:17:59 +0000: Admin server listening on port 3001
2024-06-03 16:17:59 03/Jun/2024:09:17:59 +0000: Listening on port 3000
2024-06-03 16:17:59 Assertion failed: *(&once->n) != 4 (threads.c: k5_once: 564)
2024-06-03 16:18:01 03/Jun/2024:09:18:01 +0000: Starting PostgREST 12.1...
2024-06-03 16:18:01 03/Jun/2024:09:18:01 +0000: Attempting to connect to the database...
2024-06-03 16:18:01 03/Jun/2024:09:18:01 +0000: Admin server listening on port 3001
2024-06-03 16:18:01 03/Jun/2024:09:18:01 +0000: Listening on port 3000
2024-06-03 16:18:01 Assertion failed: *(&once->n) != 4 (threads.c: k5_once: 564)
2024-06-03 16:18:05 03/Jun/2024:09:18:05 +0000: Starting PostgREST 12.1...
2024-06-03 16:18:05 03/Jun/2024:09:18:05 +0000: Attempting to connect to the database...
2024-06-03 16:18:05 03/Jun/2024:09:18:05 +0000: Admin server listening on port 3001
2024-06-03 16:18:05 03/Jun/2024:09:18:05 +0000: Listening on port 3000
2024-06-03 16:18:05 Assertion failed: *(&once->n) != 4 (threads.c: k5_once: 564)
2024-06-03 16:18:12 03/Jun/2024:09:18:12 +0000: Starting PostgREST 12.1...
2024-06-03 16:18:12 03/Jun/2024:09:18:12 +0000: Attempting to connect to the database...
2024-06-03 16:18:12 03/Jun/2024:09:18:12 +0000: Admin server listening on port 3001
2024-06-03 16:18:12 03/Jun/2024:09:18:12 +0000: Listening on port 3000
2024-06-03 16:18:12 Assertion failed: *(&once->n) != 4 (threads.c: k5_once: 564)
```

Which makes Next.js unable to produce development output when developing using `pnpm dev`, and returns the following error in the console:
```sql
{
  code: 'PGRST002',
  details: null,
  hint: null,
  message: 'Could not query the database for the schema cache. Retrying.',
  digest: '4171135552'
}
```
and the error that prevented the development build output:
```sql
{
  message: 'An invalid response was received from the upstream server',
  digest: '1306904865'
}
```

**System information**
 - Version of OS: Windows 11 23H2 (Build 22631.3672)
 - Version of CLI: 1.172.2
 - Version of Docker: v4.30.0
 - Versions of services:
 
 ```
        SERVICE IMAGE      │        LOCAL         │  LINKED
  ─────────────────────────┼──────────────────────┼────────────
    supabase/postgres      │ 15.1.1.56            │ 15.1.1.56
    supabase/gotrue        │ v2.152.0             │ v2.152.0
    postgrest/postgrest    │ v12.1                │ v12.1
    supabase/realtime      │ v2.28.32             │ -
    supabase/storage-api   │ v1.2.1               │ v1.2.1
    supabase/edge-runtime  │ v1.53.3              │ -
    supabase/studio        │ 20240422-5cf8f30     │ -
    supabase/postgres-meta │ v0.80.0              │ -
    supabase/logflare      │ 1.4.0                │ -
    bitnami/pgbouncer      │ 1.20.1-debian-11-r39 │ -
    darthsim/imgproxy      │ v3.8.0               │ -
```

### PR DESCRIPTION
**Pull request by steve-chavez**
Release notes at https://github.com/PostgREST/postgrest/releases/tag/v12.2.0

Fixes https://github.com/supabase/cli/issues/2361

### CODE REVIEW

**Review by thebengeu — APPROVED** [review]

**Review by samrose — APPROVED** [review]

File: ansible/vars.yml [hunk 1]
```diff
@@ -10,11 +10,12 @@ postgresql_release_checksum: sha256:ea2cf059a85882654b989acd07edc121833164a30340
 pgbouncer_release: "1.19.0"
 pgbouncer_release_checksum: sha256:af0b05e97d0e1fd9ad45fe00ea6d2a934c63075f67f7e2ccef2ca59e3d8ce682
 
-# to get these use `wget https://github.com/PostgREST/postgrest/releases/download/v12.0.2/postgrest-v12.0.2-ubuntu-aarch64.tar.xz -q -O- | sha1sum`
-# note: update temporarily hardcoded value over at docker/all-in-one/Dockerfile on the next PostgREST version
```
Comment (darora) [inline ansible/vars.yml:14]: The Dockerfile needs to be updated per this comment.

Simplest way would be to revert c318ad8ce966ef84d31fc27412f95917b7d53292.

File: ansible/vars.yml [hunk 2]
```diff
@@ -10,11 +10,12 @@ postgresql_release_checksum: sha256:ea2cf059a85882654b989acd07edc121833164a30340
...
-postgrest_release: "12.0.2-listeners-alpha"
-postgrest_arm_release_checksum: sha1:a61633a4118eaefd5351ad236744fd84fbb7886e
-postgrest_x86_release_checksum: sha1:d57eecdf732b7fd2a38889b35f1ed484a719d003
+# to get these use
+# wget https://github.com/PostgREST/postgrest/releases/download/v12.2.0/postgrest-v12.2.0-ubuntu-aarch64.tar.xz -q -O- | sha1sum
+# wget https://github.com/PostgREST/postgrest/releases/download/v12.2.0/postgrest-v12.2.0-linux-static-x64.tar.xz -q -O- | sha1sum
```
Comment (darora) [inline ansible/vars.yml:15]: Duplicate line, remove

File: ansible/vars.yml [hunk 3]
```diff
@@ -10,11 +10,12 @@ postgresql_release_checksum: sha256:ea2cf059a85882654b989acd07edc121833164a30340
...
+postgrest_release: "12.2.0"
```
Comment (darora) [inline ansible/vars.yml:16]: https://github.com/supabase/postgres/commit/64260decf21c1163106a66c5f86255251934397e needs to be reverted

**Review by darora — CHANGES_REQUESTED** [review]

**Review by olirice — APPROVED** [review]

File: ansible/vars.yml (hunk 1 above)
Comment (steve-chavez) [reply ansible/vars.yml:14]: Done. Reverted in https://github.com/supabase/postgres/pull/1003/commits/a502c44a100e7283ad0997ea082d03c33d623f63

File: ansible/vars.yml (hunk 3 above)
Comment (steve-chavez) [reply ansible/vars.yml:16]: Reverted in https://github.com/supabase/postgres/pull/1003/commits/6d01ddb9946e4e2a83108c19f5b1800352dfc035

**Review by darora — APPROVED** [review]

**Pull request merged**